### PR TITLE
feat(QueueCommand): Pagination

### PR DESCRIPTION
**Pull request by lxndr-rl**
- [x] Bot removes all reactions other than ◀️ | ▶️.

- [x] Bot remove user.reaction (for a new use)

- [x] Bot delete queue message when it `end`

Use only **1** ReactionCollector

fixQueue

### CODE REVIEW

File: src/commands/QueueCommand.ts [hunk 1]
```diff
@@ -21,23 +21,45 @@ export class QueueCommand extends BaseCommand {
...
-                    msg.createReactionCollector((reaction, user) => reaction.emoji.name === "▶️" && user.id === message.author.id, { time: 80 * 1000 }).on("collect", () => {
-                        if (index + 1 === indexes.length) return undefined;
-                        index++;
-                        embed.setDescription(indexes[index]).setFooter(`Page ${index + 1} of ${indexes.length}`, "https://raw.githubusercontent.com/zhycorp/disc-11/main/.github/images/info.png");
-                        msg.edit(embed).catch(e => this.client.logger.error("QUEUE_CMD_ERR:", e));
-                    });
-                }).catch(e => this.client.logger.error("QUEUE_CMD_ERR:", e));
+                void msg.react("◀️");
+                void msg.react("▶️");
+                const filter = (reaction: any, user: any) => user.id !== msg.client.user?.id;
```
Comment (mzrtamp) [inline src/commands/QueueCommand.ts]: Missing return type on function at line 29

**Review by mzrtamp — CHANGES_REQUESTED** [review]

File: src/commands/QueueCommand.ts (hunk 1 above)
Comment (lxndr-rl) [reply src/commands/QueueCommand.ts]: According with discord.js docs `.createReactionCollector` is defined

```javascript
// Create a reaction collector
const filter = (reaction, user) => reaction.emoji.name === '👌' && user.id === 'someID';
const collector = message.createReactionCollector(filter, { time: 15000 });
collector.on('collect', r => console.log(`Collected ${r.emoji.name}`));
collector.on('end', collected => console.log(`Collected ${collected.size} items`));
```

https://discord.js.org/#/docs/main/stable/class/Message?scrollTo=createReactionCollector

File: src/commands/QueueCommand.ts [hunk 2]
```diff
@@ -21,23 +21,45 @@ export class QueueCommand extends BaseCommand {
...
-                    msg.createReactionCollector((reaction, user) => reaction.emoji.name === "▶️" && user.id === message.author.id, { time: 80 * 1000 }).on("collect", () => {
-                        if (index + 1 === indexes.length) return undefined;
-                        index++;
-                        embed.setDescription(indexes[index]).setFooter(`Page ${index + 1} of ${indexes.length}`, "https://raw.githubusercontent.com/zhycorp/disc-11/main/.github/images/info.png");
-                        msg.edit(embed).catch(e => this.client.logger.error("QUEUE_CMD_ERR:", e));
-                    });
-                }).catch(e => this.client.logger.error("QUEUE_CMD_ERR:", e));
+                void msg.react("◀️");
+                void msg.react("▶️");
+                const filter = (reaction: any, user: any) => (reaction.emoji.name === "◀️" || reaction.emoji.name === "▶️") && user.id !== msg.client.user?.id;
```
Comment (Allvaa) [inline src/commands/QueueCommand.ts]:
```suggestion
                const filter = (reaction: any, user: any): boolean => (reaction.emoji.name === "◀️" || reaction.emoji.name === "▶️") && user.id !== msg.client.user?.id;
```
to remove warning as @mzrtamp mentioned above
image

**Review by Allvaa — CHANGES_REQUESTED** [review]

File: src/commands/QueueCommand.ts (hunk 2 above)
Comment (mzrtamp) [reply src/commands/QueueCommand.ts]: Thanks

**Comment by mzrtamp** [comment]
That's all? @lxndr-rl

**Comment by lxndr-rl** [comment]
> That's all? @lxndr-rl 

For this command I would say yes. Would it be okay if I open another PR to implement #264 ?

**Comment by mzrtamp** [comment]
> > That's all? @lxndr-rl
> 
> For this command I would say yes. Would it be okay if I open another PR to implement #264 ?

Sure, go ahead :)

**Review by mzrtamp — APPROVED** [review]
Thanks

**Pull request merged**